src/OSPSuite.InstallationValidator.Core/Domain/StartableProcess.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading;
+using Microsoft.Win32.SafeHandles;
+
+namespace OSPSuite.InstallationValidator.Core.Domain
+{
+   //TODO - move class to core
+   public class StartableProcess : IDisposable
+   {
+      private readonly Process _process;
+      private bool _exited;
+
+      public StartableProcess(string filePath, params string[] arguments)
+      {
+         _process = new Process
+         {
+            StartInfo = new ProcessStartInfo
+            {
+               FileName = filePath,
+               Arguments = string.Join(" ", arguments)
+            }
+         };
+      }
+
+      /// <summary>
+      /// <returns>true</returns> if the process exited normally. If the process was stopped, returns false
+      /// </summary>
+      public bool ExitedNormally => _exited;
+
+      /// <summary>
+      /// <returns>the exit code of the process</returns> if <see cref="ExitedNormally"/> returns true.
+      /// throws an <seealso cref="InvalidOperationException"/> if the process has not exited or was aborted abnormally
+      /// </summary>
+      public int ReturnCode => _process.ExitCode;
+
+      public virtual void Start()
+      {
+         _process?.Start();
+      }
+
+      public void Dispose()
+      {
+         _process.Dispose();
+      }
+
+      public virtual void Wait(CancellationToken token)
```

Comment:
I have no idea what that all does here. I am assuming you tried it with the cancel and it works?